##### What is distribution of number of genes that might be compound heterozygous for LOF SNPs?
png('compound_het.png')
plot(x, pg1_comphet_y, ylim=c(0,25), type="l", col=colours[1], main="Cumulative count compound heterozygous genes\nfor LOF SNPs", xlab="MAF", ylab="Candidate gene count")
par(new=T)
plot(x, pg3_comphet_y, ylim=c(0,25), type="l", col=colours[2], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg4_comphet_y, ylim=c(0,25), type="l", col=colours[3], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg5_comphet_y, ylim=c(0,25), type="l", col=colours[4], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg6_comphet_y, ylim=c(0,25), type="l", col=colours[5], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg7_comphet_y, ylim=c(0,25), type="l", col=colours[6], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg8_comphet_y, ylim=c(0,25), type="l", col=colours[7], axes=FALSE, xlab="", ylab="")
par(new=T)
plot(x, pg9_comphet_y, ylim=c(0,25), type="l", col=colours[8], axes=FALSE, xlab="", ylab="")
legend(0, 25, lty=1, c('pg1 (Venter)','pg3 (Korean)','pg4 (Korean)','pg5 (Chinese)','pg6 (p0)','pg7 (NA19240)','pg8 (NA19240)', 'pg9 (Church)'), col=colours)
dev.off()

qplot(individual, value, data = merged_consequences, geom="bar", fill = individual, log="y", xlab = NULL) + facet_grid(. ~ consequence) + opts(strip.text.x = theme_text(angle=90, size=5)) + opts(axis.text.x = theme_blank())
#qplot(individual, value, data = merged_consequences, geom="bar", fill = individual, log="y", xlab = NULL) + facet_grid(. ~ consequence) + opts(strip.text.y = theme_text(angle=90, vjust=1))
ggsave(file='merged_consequences.png', width=8, height=4)

qplot(individual, value, data = merged_relative_genotypes, geom="bar", fill = individual, xlab = NULL) + facet_grid(. ~ relative_genotype) + opts(axis.text.x = theme_blank())
ggsave(file='merged_relative_genotypes.png', width=8, height=4)

png('lof_snps.png',width=1200,height=800)
grid.newpage()
pushViewport(viewport(layout = grid.layout(2, 2)))
print(qplot(maf, count, data=lof_snps, geom="line", colour=individual, ylim=c(0,450), main="Cumulative count LOF SNPs", lwd=1), vp=vplayout(1,1))
print(qplot(maf, count, data=lof_snps_pilot1, geom="line", colour=individual, ylim=c(0,300), main="Cumulative count LOF SNPs\nalso present in 1000genomes", lwd=1), vp=vplayout(2,1))
print(qplot(individual, count, data=lof_snps_nonpilot1, geom="bar", main="LOF SNPs\nnot present in 1000genomes", lwd=1, fill=individual) + opts(axis.text.x=theme_blank()), vp=vplayout(2,2))
dev.off()

qplot(nr_individuals, count, data=nr_individuals_per_lof_gene, geom="bar", xlab="Number individuals", ylab="Number LOF genes")
ggsave(file='nr_individuals_per_lof_gene.png', width=6, height=3)
